tcms/telemetry/api.py

```diff
@@ -226,6 +226,42 @@ def test_case_health(query=None):
     return data
 
 
+@http_basic_auth_login_required
+@rpc_method(name="Testing.individual_test_case_health")
+def individual_test_case_health(query=None):
+
+    if query is None:
+        query = {}
+
+    res = {}
+    plan_id = 0
+    positive, negative, all_count = 0, 0, 0
+    for test_execution in TestExecution.objects.filter(**query).order_by(
+        "run__plan_id"
```

Comment:
Order by status__weight as well. I think this will have a small positive effect towards performance b/c it will be more likely that the same if branches below will be executed multiple times which leads to better chances of taking advantage of some parallelization/caching on the OS/CPU level.